.github/actions/npm-blacklist/checker.sh

```
@@ -0,0 +1,105 @@
+    
+    # Parse package name and version
+    if [[ "$line" =~ ^([^@[:space:]]+)[@[:space:]]+(.+)$ ]]; then
+        pkg_name="${BASH_REMATCH[1]}"
+        pkg_version="${BASH_REMATCH[2]}"
+    else
+        echo "Warning: Invalid format in line: $line"
+        continue
+    fi
+    
+    # Remove any trailing whitespace from version
+    pkg_version=$(echo "$pkg_version" | sed 's/[[:space:]]*$//')
```

Shellcheck (0.10.0)

[style] See if you can use ${variable//search/replace} instead.

(SC2001)